specs/0118-notification-center.md

```diff
@@ -0,0 +1,119 @@
+  - Checkbox (default empty): `This is a transactional email`
+  - Text description: `Transactional emails are sent to users who have not opted in to email communication. Transactional emails are restricted in the type of content that can be included.`
+  - Validation:
+    - Error if `Transactional email` enabled but `Message type` does not include `Email`: `You cannnot enable the transactional email option if the Message type does not include email.`
+- Button: `Review message`
+  - A send button is not displayed on the same page as the message editor and recipient selection to avoid sending an incomplete message or a message to the wrong recipients.
+  - When pressed a validation step is performed on inputs to ensure there are no issues with the form. If there are no issues, the user proceeds to the Review/Confirmation page.
+
+- Title: `Review message`
+- A review step is required to reduce errors because messages could be sent to a large number of people. This page allows you to review all aspects of the email before sending, including:
+ - A rendered version of the message to ensure things like Markdown tags or html don’t have issues.
+ - A summary of who will be receiving the nessage:
+    - The number of recipients
+    - The filters being applied
+  - Whether this is a transactional email. If enabled, a warning will be shown: `Warning: transactional emails are sent to users who have not opted in to email communication. Transactional emails are restricted in the type of content that can be included. When in doubt, please review with legal.`
+  - Button: `Edit`
+    - Clicking the `Edit` button or navigating back in the browser brings you to the previous page
+    - The previously entered content is preserved when `Edit` or the browser's back button is useds so work is not lost
+  - Checkbox (default empty): `I’ve reviewed the work and this is ready to be sent.`
+    - This is another safeguard to avoid accidentally sending messages without thorough review.
+    - Validation:
+      - Error if this is not checked: `You must confirm that you have reviewed the contents of this message.`
+  - Button: `Send`
+    - Triggers validation step
+    - Initiates the sending of the email and/or notification
+
+## Email Content
+
+When sending an email to users via the Notification Center, emails include the following content:
+- Subject: Content filled from “Subject” field in Notification Center
+- Body: Content filled from “Body” field in Notification Center (rendered in email compliant HTML)
+- Footer: Added to the bottom of all non-transactional emails: `You’re receiving this email as a contributor to Mozilla localization on Pontoon. <br>To no longer receive emails like these, unsubscribe here: <unsubscribe-link>Unsubscribe</unsubscribe-link>.`
+
+## Other
```

Comment:
Are we going to remove the ability to see messages that have already been sent? Keeping the list of sent items would let us recycle and even make changes (in case of in-app messages).

Reply:
I had considered this and thought the scope of implementing a feature like this might be too big and opted for a MVP. If you don't consider that too large, I'll add it to this spec.
(Also do you have any thoughts @flodolo based on our teams overall state in terms of bandwidth?)

Reply:
I agree that we shouldn't include this in the first iteration:
* In-app notifications: right now they're duplicated across the DB (one copy for recipient). Are you thinking of refactoring notifications as a preliminary step for this feature? Otherwise, I don't see how you would be able to correct a past notification.
* I expect more complex messages to live in documents, if they need to be discussed.

Reply:
For pretty much every email I have sent from Pontoon we had to chase Google Docs to see what exactly was the target audience for a previous similar email. Having all that stored in one place would be very helpful, and it's not that hard to implement, given we already have a similar functionality.

We can edit in-app notifications without changing the model (although we should do that, too). But I would put the editing part in the [Out of scope](https://github.com/mozilla/pontoon/blob/main/specs/0000-template.md#optional-sections) section. With all the Review capabilities I also hope there will be less need to actually edit notifications.

Reply:
So what's your suggestion to add here? Specifically, what are you imaging when you say it's not hard to implement/have similar functionality?

Reply:
"Inbox" - the ability to see messages that have already been sent.

Reply:
@mathjazz - I've added a basic idea of what this "inbox" could be. Could you add suggestions to this to align with what you were envisioning?